keyboards/cantor/cantor.c

```diff
@@ -0,0 +1,11 @@
+// Copyright 2023 Alessio Caiazza (@nolith)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "quantum.h"
+
+// Workaround for https://github.com/qmk/qmk_firmware/issues/16406
+__attribute__ ((weak))
+void suspend_wakeup_init_kb(void) {
+    NVIC_SystemReset();
```

Comment:
This isn't a fix. That's a hard reset...

You'd be better off just adding `QK_BOOT` to your keymap.

Reply:
Thank you @drashna for the review. Could you expand a bit more on how adding a `QK_BOOT` is better than this?

As I wrote in the description this is not a fix but a workaround.

With a 36 keys layout there is no space for a `QK_BOOT` on the first layer, and the bug itself, that blocks the dual function keys will make it impossible to have activate it from another layer.

I already shared in the PR description that my knowledge of the QMK internals is very poor, so I think I may be missing something here. Could you please explain me why an hard reset is so problematic with a simple design like the cantor that has no fancy features (no RGB, no diodes, no speakers, no rotary encoders).

Thank you

Reply:
Force rebooting when it wakes is no t the right way to handle this, at all.  This may actually cause issues for other users that are not experiencing the issue.

#19780 is aimed at fixing the issue correctly, in core.

In the meanwhile, `NO_SUSPEND_POWER_DOWN = yes` in your rules.mk would likely solve this issue, as would `#define USB_SUSPEND_WAKEUP_DELAY 200` in your config.h.

Reply:
Thanks for explaining, I'll close this PR